lang/java/reef-webserver/src/main/java/org/apache/reef/webserver/HttpServerImpl.java

```diff
@@ -76,17 +82,18 @@
    * @throws Exception
    */
   @Inject
-  HttpServerImpl(final JettyHandler jettyHandler,
-                 final LocalAddressProvider addressProvider,
+  HttpServerImpl(@Parameter(RemoteConfiguration.HostAddress.class) final String hostAddress,
                  @Parameter(TcpPortRangeBegin.class)final int portNumber,
+                 final JettyHandler jettyHandler,
+                 final LocalAddressProvider addressProvider,
                  final TcpPortProvider tcpPortProvider,
                  final LoggingScopeFactory loggingScopeFactory) throws Exception {
-    this.addressProvider = addressProvider;
     this.loggingScopeFactory = loggingScopeFactory;
     this.jettyHandler = jettyHandler;
     int availablePort = portNumber;
     Server srv = null;
 
+    this.hostAddress = UNKNOWN_HOST_NAME.equals(hostAddress) ? addressProvider.getLocalAddress() : hostAddress;
     try (final LoggingScope ls = this.loggingScopeFactory.httpServer()) {
 
       final Iterator<Integer> ports = tcpPortProvider.iterator();
```

Comment:
maybe something like
```java
Server srv = null;
int availablePort = -1;
for (final int p : tcpPortProvider) {
  if ((srv = tryPort(p)) != null) {
    availablePort = p;
    break;
  }
}

```

Reply:
I had to tweak your suggestion slight to avoid a check style violation: "InnerAssignment: Inner assignments should be avoided.". Other than that, I used your code as is.

---
In reply to: [207104686](https://github.com/apache/reef/pull/1478#discussion_r207104686) [](ancestors = 207104686)